internal/server/middleware.go

```diff
@@ -157,6 +158,7 @@ func unauthenticatedMiddleware(srv *Server) gin.HandlerFunc {
 				sendAPIError(c, internal.ErrBadRequest)
 				return
 			}
+			authned.Organization = org
```

Comment:
This is the unauthenticated route, so it may not be set yet.

I wasn't sure where to store it. Maybe we should split up the `Authenticated` struct into something like this:

```go
type RequestDetails struct {
    Authenticated Authenticated
    Organization *models.Organization
}
```

Reply:
That seems like a good idea